crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -404,6 +407,44 @@ impl SettlementEncoder {
         }
     }
 
+    // In case there are ethflow orders, we need a pre_interaction that calls
+    // the wrapAll() funcitonality to wrap all the ether used for ethflow orders
+    // to WETH
+    fn build_ethflow_pre_interactions(&self) -> Vec<EncodedInteraction> {
+        let mut ethflow_orders_owners: Vec<H160> = self
+            .liquidity_order_trades
+            .iter()
+            .filter_map(|order_trade| {
+                if order_trade.trade.order.metadata.is_ethflow_order {
+                    Some(order_trade.trade.order.metadata.owner)
+                } else {
+                    None
+                }
+            })
+            .collect();
+        ethflow_orders_owners.append(
+            &mut self
+                .order_trades
+                .iter()
+                .filter_map(|order_trade| {
+                    if order_trade.trade.order.metadata.is_ethflow_order {
+                        Some(order_trade.trade.order.metadata.owner)
+                    } else {
+                        None
+                    }
+                })
+                .collect(),
+        );
```

Comment:
nit: I think we shouldn't collect + iterate + get, but instead use `find_map` here:
```rust
        let ethflow_owner: Option<H160> = self.liquidity_order_trades
            .iter()
            .map(|order_trade| &order_trade.trade)
            .chain(
                self.order_trades
                    .iter()
                    .map(|order_trade| &order_trade.trade),
            )
            .find_map(|trade| {
                if trade.order.metadata.is_ethflow_order {
                    Some(trade.order.metadata.owner)
                } else {
                    None
                }
            });
```